ssh/tailssh/incubator.go

```diff
@@ -244,22 +245,10 @@ func beIncubator(args []string) error {
 		groupIDs = append(groupIDs, int(gid))
 	}
 
-	if err := setGroups(groupIDs); err != nil {
+	if err := dropPrivileges(logf, int(ia.uid), ia.gid, groupIDs); err != nil {
```

Comment:
Hmm, why is ia.uid an int64 but gids are a regular int? Whenever I see a cast to int it sets off my integer overflow spidey sense. I wonder if we can standardize on whichever type the syscalls want, so that the parsing funcs take care of overflow handling? (_do_ they take care of overflow handling?)

Reply:
@maisem - do you remember why? On every platform that I can find (Linux, Darwin, various BSDs), `uid_t` is 32-bit, and the Go stdlib takes the `int` type for both `os.Getuid` and the `syscall.{Get,Set}uid` functions.

Reply:
I went back to find the history and couldn't this looks like an oversight.